src/wp-includes/class-wp-embed.php

```diff
@@ -30,6 +30,7 @@ class WP_Embed {
 	 */
 	public function __construct() {
 		// Hack to get the [embed] shortcode to run before wpautop().
+		add_filter( 'the_block_template_html', array( $this, 'run_shortcode' ), 6 );
```

Comment:
_If_ we go with my other suggestion on adjusting the shortcode filter priorities, we could align this:
```suggestion
		add_filter( 'the_block_template_html', array( $this, 'run_shortcode' ), 8 );
```